src/flb_network.c

```diff
@@ -267,10 +268,9 @@ static int net_connect_sync(int fd, const struct sockaddr *addr, socklen_t addrl
         FD_ZERO(&wait_set);
         FD_SET(fd, &wait_set);
 
-        /* Wait 'connect_timeout' seconds for an event */
-        timeout.tv_sec = connect_timeout;
-        timeout.tv_usec = 0;
-        ret = select(fd + 1, NULL, &wait_set, NULL, &timeout);
+        pfd_read.fd = fd + 1;
+        pfd_read.events = POLLIN;
+        ret = poll(&pfd_read, 1, connect_timeout) != 1;
```

Comment:
we will need an approach for other OSes... (Windows and potentially MacOS)

Reply:
Notice that  poll(2) timeout is in milliseconds.
To make compatible with select version..
Something like this should be done.. 
```c
 ret = poll(&pfd_read, 1, connect_timeout * 1000) != 1;
```